source/date_time.c

```diff
@@ -747,11 +754,12 @@ double aws_date_time_as_epoch_secs(const struct aws_date_time *dt) {
 }
 
 uint64_t aws_date_time_as_nanos(const struct aws_date_time *dt) {
-    return (uint64_t)dt->timestamp * AWS_TIMESTAMP_NANOS;
+    return ((uint64_t)dt->timestamp * AWS_TIMESTAMP_NANOS) +
```

Comment:
Why not use the time conversion functions?

Reply:
I just assumed there was a reason why the original code didn't use them everywhere either. Maybe for performance reasons?